libs/Theseus/src/Theseus.jl

```diff
@@ -1,320 +1,432 @@
 
 # Abstract supertype of Implict.jl's own time integrators for dispatch
 abstract type AbstractTimeIntegrator end
 
-import DiffEqBase
+using DiffEqBase: DiffEqBase
 
 import DiffEqBase: solve, CallbackSet, ODEProblem
 export solve, ODEProblem
 
 # Interface required by DiffEqCallbacks.jl
 function DiffEqBase.get_tstops(integrator::AbstractTimeIntegrator)
-    return integrator.opts.tstops
+	return integrator.opts.tstops
 end
 function DiffEqBase.get_tstops_array(integrator::AbstractTimeIntegrator)
-    return get_tstops(integrator).valtree
+	return get_tstops(integrator).valtree
 end
 function DiffEqBase.get_tstops_max(integrator::AbstractTimeIntegrator)
-    return maximum(get_tstops_array(integrator))
+	return maximum(get_tstops_array(integrator))
 end
 
 function finalize_callbacks(integrator::AbstractTimeIntegrator)
-    callbacks = integrator.opts.callback
-
-    return if callbacks isa CallbackSet
-        foreach(callbacks.discrete_callbacks) do cb
-            cb.finalize(cb, integrator.u, integrator.t, integrator)
-        end
-        foreach(callbacks.continuous_callbacks) do cb
-            cb.finalize(cb, integrator.u, integrator.t, integrator)
-        end
-    end
+	callbacks = integrator.opts.callback
+
+	return if callbacks isa CallbackSet
+		foreach(callbacks.discrete_callbacks) do cb
+			cb.finalize(cb, integrator.u, integrator.t, integrator)
+		end
+		foreach(callbacks.continuous_callbacks) do cb
+			cb.finalize(cb, integrator.u, integrator.t, integrator)
+		end
+	end
 end
 
 import SciMLBase: get_du, get_tmp_cache, u_modified!,
-    init, step!, check_error,
-    get_proposed_dt, set_proposed_dt!,
-    terminate!, remake, add_tstop!, has_tstop, first_tstop
+	init, step!, check_error,
+	get_proposed_dt, set_proposed_dt!,
+	terminate!, remake, add_tstop!, has_tstop, first_tstop
 
 
 # Abstract base type for time integration schemes
 abstract type SimpleImplicitAlgorithm{N} end
+abstract type NonDirect{N} <: SimpleImplicitAlgorithm{N} end
+abstract type Direct{N} <: SimpleImplicitAlgorithm{N} end
 
 stages(::SimpleImplicitAlgorithm{N}) where {N} = N
 
-struct ImplicitEuler <: SimpleImplicitAlgorithm{1} end
+struct ImplicitEuler <: NonDirect{1} end
 function (::ImplicitEuler)(res, uₙ, Δt, f!, du, u, p, t, stages, stage)
-    f!(du, u, p, t + Δt) # t = t0 + c_1 * Δt
+	f!(du, u, p, t + Δt) # t = t0 + c_1 * Δt
 
-    res .= uₙ .+ Δt .* du .- u # Δt * a_11
-    return nothing
+	res .= uₙ .+ Δt .* du .- u # Δt * a_11
+	return nothing
 end
 
 struct ImplicitMidpoint <: SimpleImplicitAlgorithm{1} end
 function (::ImplicitMidpoint)(res, uₙ, Δt, f!, du, u, p, t, stages, stage)
-    # Evaluate f at midpoint: f((uₙ + u)/2, t + Δt/2)
-    # Use res for a temporary allocation (uₙ .+ u) ./ 2
-    uuₙ = res
-    uuₙ .= 0.5 .* (uₙ .+ u)
-    f!(du, uuₙ, p, t + 0.5 * Δt)
-
-    res .= uₙ .+ Δt .* du .- u
-    return nothing
+	# Evaluate f at midpoint: f((uₙ + u)/2, t + Δt/2)
+	# Use res for a temporary allocation (uₙ .+ u) ./ 2
+	uuₙ = res
```

Comment:
Did I write this with tab instead of spaces?

Reply:
Especially since https://github.com/vchuravy/Ariadne.jl/pull/32/files#diff-685b366be89ca838175dba97834e1e168a90fed49f7e60c31ff06df6c3eff619 seems to change some of these back?